src/app/services/share.js

```diff
@@ -132,6 +122,115 @@ module.exports = (Model, App) => {
     return newShare.token;
   };
 
+  /**
+   * Generates a share folder token
+   * @param user
+   * @param folderId
+   * @param bucket
+   * @param mnemonic
+   * @param fileToken
+   * @param views
+   * @returns {Promise<string|*>}
+   * @constructor
+   */
+  const GenerateFolderTokenAndCode = async (
+    user,
+    folderId,
+    bucket,
+    mnemonic,
+    fileToken,
+    views = 1,
+  ) => {
+    const itemExists = await Model.folder.findOne({
+      where: {
+        id: { [Op.eq]: folderId },
+        user_id: { [Op.eq]: user.id },
+      }
+    });
+
+    if (!itemExists) {
+      throw Error('Folder not found');
+    }
+
+    const maxAcceptableSize = 1024 * 1024 * 1000; // 1GB
+
+    const folderSize = await getFolderSize(folderId);
+
+    if (folderSize > maxAcceptableSize) {
+      throw Error('Folder too large');
+    }
+
+    // Generate password
+    const code = crypto.randomBytes(32).toString('hex');
+
+    // Encrypt mnemonic with password
+    const encryptedMnemonic = encrypt(mnemonic, code);
+
+    // Generate a new share token
+    const newToken = crypto.randomBytes(10).toString('hex');
+
+    const share = await Model.shares.findOne({
+      where: {
+        file: {
+          [Op.eq]: folderId
+        },
+        user: {
+          [Op.eq]: user.email
+        }
+      },
+    });
+
+    if (share) {
+      // Update share details
+      Model.shares.update(
```

Comment:
Adding an `await` here could be a good idea as a method to ensure that the share is properly updated